src/js/ia/IAPage.js

```diff
@@ -240,9 +240,43 @@
                             }
                         });
                     });
+                    
+                    // Check if the IA has a test machine or is live.
+                    // If so, enable the screenshot button.
+                    var test_machine = ia_data.live.test_machine;
+                    var example_query = ia_data.live.example_query;
+                    function enableScreenshotButton() {
+                        var $generate_screenshot = $(".generate-screenshot--button");
+                        if((test_machine && example_query) || ia_data.live.dev_milestone === "live") {
+                            $generate_screenshot.removeClass("generate-screenshot--disabled");
+                        } else {
+                            $generate_screenshot.addClass("generate-screenshot--disabled");
+                        }
+                    }
+
+                    enableScreenshotButton();
+                    $("body").on("change", ".test_machine", function(evt) {
```

Comment:
There's no need to create two new events for calling the enableScreenshotButton function: you can check inside the autocommit function (somewhere [here](https://github.com/duckduckgo/community-platform/blob/jag/disable-screenshot/src/js/ia/IAPage.js#L752)) if the saved field was either the test machine or the example query, and in that case, call the function.